code/_onclick/click_alt.dm

```diff
@@ -0,0 +1,54 @@
+/**
+ * Base proc for alt click interaction.
+ *
+ * If you wish to use custom alt_click behavior, use that proc.
+ * Override this if you want to change all alt-click behavior for a type.
+ */
+/atom/proc/base_click_alt(mob/user)
+	SHOULD_CALL_PARENT(TRUE)
+	PROTECTED_PROC(TRUE)
+
+	if(!can_interact(user))
+		return
+
+	var/early_sig_return = SEND_SIGNAL(src, COMSIG_CLICK_ALT, user) & COMPONENT_CANCEL_CLICK_ALT
```

Comment:
IMO you should dump `COMPONENT_CANCEL_CLICK_ALT` and just let the signal return `CLICK_ACTION_SUCCESS` or `CLICK_ACTION_BLOCKING`